src/EnergyPlus/WaterThermalTanks.cc

```diff
@@ -3641,126 +3642,173 @@ bool getWaterTankMixedInput(EnergyPlusData &state)
+                ErrorsFound = true;
+            }
 
-        if (state.dataIPShortCut->rNumericArgs(6) > 0.0) {
-            Tank.TankTempLimit = state.dataIPShortCut->rNumericArgs(6);
-        } else {
-            // default to just above freezing
-            Tank.TankTempLimit = 1.0;
+            auto const &bottomTempScheduleFieldName = "bottom_setpoint_temperature_schedule_name";
+            auto const &bottomSetpointTempScheduleName = fields.find(bottomTempScheduleFieldName);
+            if (bottomSetpointTempScheduleName == fields.end() || bottomSetpointTempScheduleName->empty()) {
+                ShowSevereEmptyField(state, eoh, bottomTempScheduleFieldName);
+                ErrorsFound = true;
+            } else if ((Tank.setptTempSchedBottom =
+                            Sched::GetSchedule(state, Util::makeUPPER(bottomSetpointTempScheduleName.value().get<std::string>()))) == nullptr) {
+                ShowSevereItemNotFound(state, eoh, bottomTempScheduleFieldName, bottomSetpointTempScheduleName.value().get<std::string>());
+                ErrorsFound = true;
+            }
+            Tank.HeaterHeight1 = state.dataInputProcessing->inputProcessor->getRealFieldValue(fields, schemaProps, "top_temperature_sensor_height");
+            Tank.TempSensorHeight1 = Tank.HeaterHeight1;
+            Tank.HeaterHeight2 =
+                state.dataInputProcessing->inputProcessor->getRealFieldValue(fields, schemaProps, "bottom_temperature_sensor_height");
+            Tank.TempSensorHeight2 = Tank.HeaterHeight2;
+            // use side
+            auto const &UseFlowDirectionScheName = fields.find("use_side_flow_direction_schedule");
+            if (UseFlowDirectionScheName == fields.end() || UseFlowDirectionScheName->empty()) {
+                Tank.UseFlowDirectionSched = Sched::GetScheduleAlwaysOn(state);
+            } else if ((Tank.UseFlowDirectionSched =
+                            Sched::GetSchedule(state, Util::makeUPPER(UseFlowDirectionScheName.value().get<std::string>()))) == nullptr) {
+                ShowSevereItemNotFound(state, eoh, "use_side_flow_direction_schedule", UseFlowDirectionScheName.value().get<std::string>());
+                ErrorsFound = true;
+            }
+            kwHeatingCooling = "heating";
         }
 
-        if (state.dataIPShortCut->lAlphaFieldBlanks(3)) {
-            ShowSevereEmptyField(state, eoh, state.dataIPShortCut->cAlphaFieldNames(3));
-            ErrorsFound = true;
-        } else if ((Tank.setptTempSched = Sched::GetSchedule(state, state.dataIPShortCut->cAlphaArgs(3))) == nullptr) {
-            ShowSevereItemNotFound(state, eoh, state.dataIPShortCut->cAlphaFieldNames(3), state.dataIPShortCut->cAlphaArgs(3));
-            ErrorsFound = true;
+        if (TankTempLimit > 0.0) {
+            Tank.TankTempLimit = TankTempLimit;
+        } else {
+            // Default to very large number, boiling point of water
+            Tank.TankTempLimit = 100.0;
```

Comment:
For CW this is a minimum tank temperature limit and for HW this is a maximum limit. I think this needs to get wrapped by the tanktype to apply the proper context.

     N6 , \field Minimum Temperature Limit         // CW

     N7 , \field Maximum Temperature Limit        // HW

Reply:
indeed! I just added different temperature limit defaults for the hot water and chilled water tank separately.